lib/github.com/mpllang/mpllib/ArraySequence.sml

```diff
@@ -118,10 +122,31 @@ struct
     end
 
 
+  fun foldl f b s =
+    SeqBasis.foldl f b (0, length s) (nth s)
+
+  fun foldr f b s =
+    SeqBasis.foldr f b (~1, length s-1) (nth s)
```

Comment:
Recommendation: do range `(0, length s)`. See comments about `foldr` in `SeqBasis`.